packages/name-controller/src/NameController.ts

```diff
@@ -612,4 +614,34 @@ export class NameController extends BaseController<
       delete proposedNames[dormantSourceId];
     }
   }
+
+  #cleanupExpiredEntries(): void {
+    const currentTime = this.#getCurrentTimeSeconds();
+
+    this.update((state: NameControllerState) => {
+      for (const type of Object.keys(state.names)) {
+        const nameType = type as NameType;
+        for (const value of Object.keys(state.names[nameType])) {
+          for (const variation of Object.keys(state.names[nameType][value])) {
+            const entry = state.names[nameType][value][variation];
+
+            if (entry.name !== null) {
+              continue;
+            }
+
+            const proposedNames = Object.values(entry.proposedNames);
+            const allProposedNamesExpired = proposedNames.every(
+              (proposedName: ProposedNamesEntry) =>
+                currentTime - (proposedName.lastRequestTime ?? 0) >
```

Comment:
Minor, but should it be `>=` since we technically want to delete as it reaches 24 hours?

Reply:
Makes sense added in https://github.com/MetaMask/core/pull/3748/commits/0550b4af5f2f28cf95640b8b3643ce6e06f76844